core/capabilities/ccip/ccipsolana/commitcodec.go

```diff
@@ -25,16 +25,18 @@ func NewCommitPluginCodecV1() *CommitPluginCodecV1 {
 func (c *CommitPluginCodecV1) Encode(ctx context.Context, report cciptypes.CommitPluginReport) ([]byte, error) {
 	var buf bytes.Buffer
 	encoder := agbinary.NewBorshEncoder(&buf)
-	if len(report.MerkleRoots) != 1 {
-		return nil, fmt.Errorf("unexpected merkle root length in report: %d", len(report.MerkleRoots))
+	// Solana doesn't have blessed merkle root at the moment
+	if len(report.UnblessedMerkleRoots) != 1 {
+		return nil, fmt.Errorf("unexpected merkle root length in report: %d", len(report.UnblessedMerkleRoots))
 	}
 
```

Comment:
I think you wanna add a length check to `UnblessedMerkleRoots`? Or don't care about that?

Reply:
If there's a blessed root, is it an error?

Reply:
Oops yeah typo, meant `BlessedMerkleRoots`

Reply:
something like this
```
combined := append(report.UnblessedMerkleRoots, report.BlessedMerkleRoots)
if len (combined) != 1 { ... error ... }
merkleRoot := combined[0]
```

Reply:
I added code to handle rmn signatures, and blessed and unblessed roots, both on encoding and decoding side.
Please review again.